### ISSUE TITLE
Add dropout regularization

### ISSUE THREAD

**Issue by MateuszSzymanski2912** [MateuszSzymanski2912/PikminFlow#2]
Dropout regularization is a technique used in neural networks to prevent overfitting. It works by randomly "dropping out" or deactivating a subset of neurons during each training iteration. This means that during each forward and backward pass, some neurons are temporarily ignored, forcing the network to learn more robust and redundant representations. Dropout regularization adds new hyperparameter p, which is a probability of (temporarily) setting weight to 0. Each layer can have their own dropout parameter
User should be able to input:
- dropout hyperparameter (p)
